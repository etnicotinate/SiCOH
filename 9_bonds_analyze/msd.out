# Time-averaged data for fix myMSD
# TimeStep c_msd1[1] c_msd1[2] c_msd1[3] c_msd1[4] c_thermo_temp
100 0.0152707 0.0125048 0.0153655 0.043141 186.905
200 0.0652299 0.0538394 0.0697839 0.188853 248.871
300 0.119964 0.120763 0.169485 0.410212 302.665
400 0.18527 0.244088 0.305498 0.734857 338.894
500 0.299511 0.349375 0.331309 0.980195 342.042
600 0.380877 0.35044 0.432425 1.16374 323.764
700 0.403178 0.430275 0.509834 1.34329 320.708
800 0.464491 0.527297 0.581094 1.57288 317.984
900 0.478418 0.597051 0.643618 1.71909 304.516
1000 0.496913 0.665981 0.671437 1.83433 323.049
1100 0.535691 0.609242 0.713759 1.85869 308.466
1200 0.547174 0.637289 0.722609 1.90707 304.586
1300 0.539535 0.627533 0.706533 1.8736 292.253
1400 0.535656 0.598572 0.760622 1.89485 308.08
1500 0.544095 0.627969 0.7821 1.95416 320.89
1600 0.550778 0.629546 0.798644 1.97897 320.097
1700 0.56533 0.653744 0.765465 1.98454 313.257
1800 0.617178 0.729736 0.794095 2.14101 315.338
1900 0.672407 0.779152 0.877669 2.32923 339.499
2000 0.798024 0.810035 0.897245 2.5053 320.123
2100 0.955762 0.872214 0.919608 2.74758 339.848
2200 0.998298 0.882571 0.913103 2.79397 347.748
2300 1.06025 1.0215 0.933221 3.01497 308.573
2400 1.18674 1.00709 0.920435 3.11426 326.076
2500 1.26621 0.954291 0.944486 3.16498 287.288
2600 1.22775 0.937084 1.01109 3.17592 298.196
2700 1.17916 0.95488 1.01544 3.14948 285.123
2800 1.19034 1.00391 1.01943 3.21368 297.509
2900 1.17737 1.00731 1.00451 3.18919 309.116
3000 1.13023 1.00655 1.03475 3.17154 333.877
3100 1.17427 0.994976 1.06581 3.23505 332.363
3200 1.30203 1.0129 1.11321 3.42814 329.892
3300 1.45409 1.13072 1.10967 3.69448 317.977
3400 1.46899 1.13287 1.00588 3.60774 296.583
3500 1.36957 1.01688 1.01621 3.40266 279.739
3600 1.26547 0.955851 1.02285 3.24417 292.867
3700 1.2816 1.00195 1.01464 3.29819 324.263
3800 1.39382 1.05099 1.09394 3.53874 308.948
3900 1.43797 1.05823 1.21872 3.71492 335.045
4000 1.47274 1.10164 1.23258 3.80696 395.388
4100 1.5544 1.08144 1.21506 3.85089 347.034
4200 1.59238 1.08154 1.20694 3.88085 319.114
4300 1.6622 1.09578 1.19372 3.9517 296.567
4400 1.70571 1.12783 1.28566 4.1192 288.049
4500 1.67387 1.14999 1.42785 4.2517 296.766
4600 1.65789 1.1321 1.44854 4.23854 297.672
4700 1.74389 1.16812 1.41688 4.32889 299.445
4800 1.69249 1.15759 1.41385 4.26393 305.491
4900 1.66181 1.18277 1.43794 4.28252 308.305
5000 1.63577 1.20001 1.45274 4.28852 299.433
