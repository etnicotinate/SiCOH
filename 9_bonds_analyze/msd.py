import numpy as np
import matplotlib.pyplot as plt

def read_msd(filename):
    data = np.genfromtxt(filename, comments='#')
    return data

def plot_msd(filename):
    timesteps, msd_x, msd_y, msd_z, msd_tt, temp = read_msd(filename).T
    plt.scatter(temp, msd_x, s=.5)
    plt.scatter(temp, msd_y, s=.5)
    plt.scatter(temp, msd_z, s=.5)
    plt.scatter(temp, msd_tt, s=1.5) # total MSD
    plt.xlabel('Temperature (K)')
    plt.ylabel('MSD')
    plt.title('Mean Squared Displacement')
    plt.legend([r'$MSD_x$', r'$MSD_y$', r'$MSD_z$', r'$MSD_{total}$'])
    pngname = r'msd.png'
    plt.savefig(pngname, dpi=300)
    print(f'MSD has been plotted to {pngname}')

if __name__ == '__main__':
    filename = r'msd.out'
    plot_msd(filename)
    # plt.show()