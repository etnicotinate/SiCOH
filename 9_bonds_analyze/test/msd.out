# Time-averaged data for fix myMSD
# TimeStep MSD_X MSD_Y MSD_Z MSD_Total Temperture
0 1.17769e-30 3.42309e-31 1.09173e-31 1.62917e-30 300
1000 1.04144 1.00921 1.36328 3.41392 309.615
